src/com/goide/inspections/GoPlaceholderCountInspection.java

```diff
@@ -16,170 +16,321 @@
+        PsiElement resolved = psiReference != null ? psiReference.resolve() : null;
+        if (!(resolved instanceof GoFunctionOrMethodDeclaration)) return;
+
+        String functionName = StringUtil.toLowerCase(((GoFunctionOrMethodDeclaration)resolved).getName());
+        if (functionName == null) return;
+
+        if (GoPlaceholderChecker.isFormattingFunction(functionName)) {
+          checkPrintf(holder, o, (GoFunctionOrMethodDeclaration)resolved);
+        }
+        else if (GoPlaceholderChecker.isPrintingFunction(functionName)) {
+          checkPrint(holder, o, (GoFunctionOrMethodDeclaration)resolved);
+        }
       }
-    }
-    return -1;
+    };
   }
 
-  private static int getPlaceholders(@Nullable GoExpression argument) {
-    if (argument == null) return -1;
-
-    String placeholderText = resolve(argument);
-    if (placeholderText == null) return -1;
+  private static void checkPrint(@NotNull ProblemsHolder holder,
+                                 @NotNull GoCallExpr callExpr,
+                                 @NotNull GoFunctionOrMethodDeclaration declaration) {
 
-    Matcher placeholders = PLACEHOLDER_PATTERN.matcher(placeholderText);
-    Matcher countPlaceholders;
+    List<GoExpression> arguments = callExpr.getArgumentList().getExpressionList();
+    if (arguments.isEmpty()) return;
 
-    int indexCount = 0;
-    int maxIndexCount = 0;
+    GoExpression firstArg = arguments.get(0);
+    if (GoTypeUtil.isString(firstArg.getGoType(null))) {
+      String firstArgText = resolve(firstArg);
+      if (firstArgText != null) {
+        if (GoPlaceholderChecker.hasPlaceholder(firstArgText)) {
+          String message = "Possible formatting directive in <code>#ref</code> #loc";
+          holder.registerProblem(firstArg, message, ProblemHighlightType.GENERIC_ERROR_OR_WARNING);
+          return;
+        }
+      }
+    }
 
-    while (placeholders.find()) {
-      indexCount++;
+    // TODO florin: Check first argument for os.Std* output
+    // Ref code: https://github.com/golang/go/blob/79f7ccf2c3931745aeb97c5c985b6ac7b44befb4/src/cmd/vet/print.go#L617
 
-      countPlaceholders = INDEXED_PLACEHOLDER_PATTERN.matcher(placeholders.group());
-      while (countPlaceholders.find()) {
-        int index = StringUtil.parseInt(countPlaceholders.group().replaceAll("\\[|\\]", ""), -1);
-        if (index > 0 && index != indexCount) {
-          indexCount = index;
+    String declarationName = declaration.getName();
+    boolean isLn = declarationName != null && declarationName.endsWith("ln");
+    for (GoExpression argument : arguments) {
+      GoType goType = argument.getGoType(null);
+      if (isLn && GoTypeUtil.isString(goType)) {
+        String argText = resolve(argument);
+        if (argText != null && argText.endsWith("\\n")) {
+          String message = "Function already ends with new line #loc";
+          TextRange range = TextRange.create(argText.length() - 1, argText.length() + 1);
```

Comment:
I think I'm doing something wrong, if I do that, then the test will fail: `ERROR: Invalid range specified: (5,2); `
What's strange is that TextRange seems to be 1 indexed not 0 indexed but I might be wrong.